### PR TITLE
Use eBPF to do connection tracking reviewed and squashed

### PR DESCRIPTION
**Pull request by alepuccetti**
@iaguis @alban can you give a look to this PR? If you like my commits reorganization I will do a force push to the kinvolk repo.

### CODE REVIEW

**Comment by alepuccetti** [comment]
https://github.com/kinvolk/scope/pull/3#discussion_r90904025 is still not addressed

**Comment by alban** [comment]
> https://github.com/kinvolk/scope/pull/3#discussion_r90904025 is still not addressed

I think that's ok for now. The concern about tracefs was for Amazon Linux, which we could test later.

And the /usr/src etc. bind mounts are not there anymore, so that's good enough for me

File: probe/endpoint/ebpf.go [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+	pb := (*byte)(u)
+	b := *pb
+	if b == 0x04 {
+		byteOrder = binary.LittleEndian
+	} else {
+		byteOrder = binary.BigEndian
+	}
+}
+
+type EventType uint32
```
Comment (alban) [inline probe/endpoint/ebpf.go]: this does not build in CircleCI: https://circleci.com/gh/alepuccetti/scope/51

you should unexport it s/EventType/eventType/g

File: probe/endpoint/ebpf.go [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+		return "unknown"
+	}
+}
+
+// tcpEvent should be in sync with the struct in the ebpf maps.
+type tcpEvent struct {
+	// Timestamp must be the first field, the sorting depends on it
+	Timestamp uint64
+
+	Cpu   uint64
```
Comment (iaguis) [inline probe/endpoint/ebpf.go]: This should be `CPU` apparently (see https://circleci.com/gh/alepuccetti/scope/51)

File: backend/Dockerfile [hunk 3]
```diff
@@ -1,16 +1,20 @@
-FROM golang:1.7.1
-RUN apt-get update && \
-	apt-get install -y libpcap-dev python-requests time file shellcheck && \
-	rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+FROM alpine:edge
+ENV GOPATH /go
+ENV PATH /go/bin:/usr/lib/go-1.7/bin:/usr/bin:/bin:/usr/sbin:/sbin
+RUN echo "http://dl-cdn.alpinelinux.org/alpine/edge/community" >>/etc/apk/repositories && \
+	apk add --update bash runit conntrack-tools iproute2 util-linux curl build-base linux-headers \
+	libpcap-dev file git py-requests go && \
```
Comment (alban) [inline backend/Dockerfile]: missing python, CircleCI error:
https://circleci.com/gh/alepuccetti/scope/51

./tools/test: /go/src/github.com/weaveworks/scope/tools/sched: /usr/bin/python: bad interpreter: No such file or directory

**Comment by alepuccetti** [comment]
Closing in favour of https://github.com/kinvolk/scope/pull/3

**Pull request closed without merging**